cmp/form/FormModel.ts

```diff
@@ -274,4 +293,37 @@ export class FormModel extends HoistModel {
             }
         );
     }
+
+    private initPersist({
+        fieldsToPersist = null,
+        path = 'form',
+        ...rootPersistWith
+    }: FormPersistOptions) {
+        const fieldNameMap = fieldsToPersist || Object.keys(this.fields);
+
+        fieldNameMap.forEach(name => {
+            PersistenceProvider.create({
+                persistOptions: {
+                    path: `${path}.${name}.value`,
+                    ...rootPersistWith
+                },
+                target: {
+                    getPersistableState: () => {
+                        return new PersistableState(this.fields[name].value);
+                    },
+                    setPersistableState: ({value}) => {
+                        // There is no metadata on a field to denote it is a date.
+                        // Use a regex matcher to tests for dates and format matches accurately.
+                        if (isString(value) && value.match(/^\d{4}-\d{2}-\d{2}$/)) {
+                            this.setValues({[name]: LocalDate.from(value)});
+                            return;
+                        }
+
+                        this.setValues({[name]: value});
```

Comment:
The internal means to set a value is this setValues (plural). A form has multiple child models - one for each form field. Hence the form parent's setValues implementation iterates through all child models, comparing names to the provided name in the argument object passed into setValues.